test/browser-bundle/browser-bundle.js

```diff
@@ -36,11 +37,13 @@ class Node extends libp2p {
   constructor (peerInfo, peerBook, options) {
     options = options || {}
     const webRTCStar = new WebRTCStar()
+    const wsStar = new WebSocketStar({ id: peerInfo.id })
```

Comment:
Does it still require that the PeerInfo gets passed on the constructor? Why?

Reply:
in the README of ws-star it says: `const ws = new WSStar({ id: id }) // the id is required for the crypto challenge`

Reply:
this requierment to have the `{ id: peerInfo.id }` makes it impossible to configure the `WebSocketStar` through the IPFS node constructor like:

```
libp2p: { // add custom modules to the libp2p stack of your node
    modules: {}
  }
```

because the peer id is not available at that point

Reply:
How else can the transport get the id?

Reply:
The first question should be "why does this Id exist in the first place", if:

- a) is to auth with the rendezvous point, the reality is that the rendezvous point doesn't have any list of pre-shared ids, so using the PeerId or something else at random is the same
- b) is to auth with the other peers, then that is a double crypto challenge because secio already does that.

Reply:
It is to prevent https://github.com/libp2p/js-libp2p-webrtc-star/issues/116

Reply:
@mkg20001 I understand but note that to fix it you don't need the node peer-id, it could be some other id of the client.

Reply:
@diasdavid I already had implemented that some time ago and you rejected it. https://github.com/libp2p/js-libp2p/issues/117#issuecomment-325929963

Reply:
There are two things here:

- a) Authentication between peers (end to end) - assured by SECIO ✔️
- b) Authentication with the rendezvous point  - what you are trying to achieve with the PeerId

b) is a bit more complicated because you want to peers to prove who they are with the rendezvous point. 

I'm torn that one the best solution is through PeerId because that really breaks the transport interface expectation. We also want to move away from centralized rendezvous points and just make any node be able to do that, which kind of means that a transport will need access to libp2p itself. Once we end up doing this, it might lead to changing how modules are loaded to a DI scheme.

Reply:
There are now 4 possible solutions:
 - Use `wsstar.lazySetId()` somewhere later in the code
 - Don't add the peerId and run the server with `--disableCryptoChallenge` (*insecure!*)
 - https://github.com/libp2p/js-libp2p/issues/117#issuecomment-325929963
 - DI scheme as mentioned above (isn't implemented so this isn't a good short-term solution)
 - Something else that works short-term at least

Which one would be the best? (I prefer the first one)

Reply:
@mkg20001 what about deploying a first version without the cryptoChallenge so that js-ipfs users can try it out today and we work on the DI scheme (which @pgte and me will start using more of his time towards that) as a separate endeavor so that then we can add the crypto challenge?

Reply:
@diasdavid The peerId IS accesible at the time of transport creation. No need to use the insecure version. https://github.com/ipfs/js-ipfs/compare/master...mkg20001:master.diff

Reply:
@diasdavid Insecure server running at `/dns/ws-star-signal-4.servep2p.com/wss/p2p-websocket-star/`